components/sup/src/manager/service_updater.rs

```diff
@@ -384,67 +445,100 @@ impl Worker {
     /// Polls until a newer version of the specified package is
     /// available. When such a package is found, it is installed, and
     /// the function exits.
-    fn run_once(&mut self, sender: SyncSender<PackageInstall>, ident: PackageIdent) {
+    fn run_once(
+        &mut self,
+        sender: Sender<PackageInstall>,
+        ident: PackageIdent,
+        kill_rx: Receiver<bool>,
+    ) {
         // Fairly certain that this only gets called in a rolling update
         // scenario, where `ident` is always a fully-qualified identifier
         outputln!("Updating from {} to {}", self.current, ident);
         let install_source = (ident, *PackageTarget::active_target()).into();
-        loop {
-            let next_time = self.next_period_start();
+        let mut next_time = None::<SteadyTime>;
 
-            match util::pkg::install(
-                // We don't want anything in here to print
-                &mut UI::with_sinks(),
-                &self.builder_url,
-                &install_source,
-                &self.channel,
-            ) {
-                Ok(package) => {
-                    self.current = package.ident().clone();
-                    sender.send(package).expect("Main thread has gone away!");
+        loop {
+            match kill_rx.try_recv() {
+                Ok(_) => {
+                    info!("Received some data on the kill channel. Letting this thread die.");
+                    break;
+                }
+                Err(TryRecvError::Empty) => {}
+                Err(TryRecvError::Disconnected) => {
+                    info!("Service updater has gone away, yikes!");
```

Comment:
Given that it's "yikes"-worthy, perhaps a `warn!` (or even `error!`)?

Reply:
Will do